Assets/Scripts/Effects/Description.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Dynamic descriptions for list of effects
+// The intent is for cards to use the text 
+// in CardData.description unless they're on top
+// of the queue, when they use one of these.
+// This is a liiiittle bit of a cop-out to avoid 
+// caring about the context an effect appears in
+
+using System.Collections.Generic;
+using Effects;
+using Cards;
+
+// Describes damage, healing, armor, Deciever cost --
+// everything that happens in Effects.Effect
+public class EffectDescription
+{
+    public class Stats
+    {
+        // Healing represented by negative damage
+        public float Damage;
+        public float Armor;
+        public Stats(float a, float b)
+        { Damage = a;
+          Armor = b;
+        }
+    }
+    public Stats EnemyStats;
+    public Stats SelfStats;
+    public bool NotAmplified = false;
+    public bool DiscardDecievers = false;
+
+    public EffectDescription()
+    {
+        Initialize();
+    }
+
+    //Aggregate all the effects that appear on a single card
+    public void Update(List<Effect> EffectList, List<Card> queue, List<Card> discardPile, int cardsInHand)
+    {
+        Initialize();
+
+        foreach (var effect in EffectList)
+        {
+            if (queue.Count < effect.minStreak)
+            {
+                continue;
+            }
+
+            if (effect.affectiveUnit == UnitType.Enemy)
+            {
+                GetStats(EnemyStats, discardPile, cardsInHand, effect);
+            }
+
+            else
+            {
+                GetStats(SelfStats, discardPile, cardsInHand, effect);
+            }
+
+            NotAmplified = NotAmplified || (effect.notAmplified);
+            DiscardDecievers = DiscardDecievers || (effect.type == EffectType.DiscardDeceiver);
+        }
+    }
+
+    private void Initialize()
+    {
+        EnemyStats = new Stats(0, 0);
+        SelfStats = new Stats(0, 0);
+        NotAmplified = false;
+        DiscardDecievers = false;
+    }
+
+    // Figure out how much damage / healing / armor an effect is going to affect
+    private static void GetStats(Stats statBlock, List<Card> discardPile, int cardsInHand, Effect effect)
+    {
+        switch (effect.type)
```

Comment:
Say I added 4 more effects, I need to update this 'switch' block as well as the actual 'ability' block... not a great plan

Reply:
(A) If they're at all complicated you'd need to change this file anyway, (B) there's one week left what are we doing adding four more effects

Reply:
Programmers want to do 'general' things. Write the code in a pure 'String Parser' way would make adding and maintaining existing code much easier.

... but anyway. still checking